### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by Luceium**
Changes front end of Log in and Sign up page.
Note this relies on #1246 and #1242

### CODE REVIEW

**Comment by Luceium** [comment]
image

image

**Comment by Luceium** [comment]
closed background PR since this PR includes it.

"Adds <background/> element that can be placed with no side effects.
The background has multiple optional props to modify it as desired.
All props have defaults and are a multiplier of the size. Notable defaults will be listed.

size: changes the size of the background (defaults to min(screen height, screen width)
innerRadius: controls size of inner square
outerRadius: controls size of outer square
radius: controls how round corners are (defaults to 0)
lineSpread: controls seperation between circuit lines
endDistance: controls how far out the circuit lines go
lineStartOffset: allows circuit lines to start past the inner square
image "

**Review by danknessdra — APPROVED** [review]
LGTM

File: src/Components/Background/background.jsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  const outerRadius = props.outerRadius || size * 0.2;
+  const lineSpread = props.lineSpread || 0.4;
+  const endDistance = props.endDistance || 1.85;
+  const lineStartOffset = props.lineStartOffset || 0;
+
+  // placement variables
+  const lineStart = (innerRadius - lineStartOffset ) / Math.sqrt(2);
+  const lineMid = outerRadius / Math.sqrt(2);
+  const lineEnd = outerRadius * endDistance;
+  const spread = (lineSpread * innerRadius) / Math.sqrt(2);
```
Comment (hsorathia) [inline src/Components/Background/background.jsx:18]: nit:  not sure what the standard is in js anymore,  but it  might be worth extracting this to a helper function in this file and then returning these values so we can DRY the code up a little bit more

@evanugarte thoughts

**Review by hsorathia — COMMENTED** [review]
pretty cool feature, left a few nit comments below

just had a few questions:
1. is the purpose of this to add the bluelines to the SVG?
2. if the above is yes, then why not duplicate/modify the SVG?
3. is the plan to animate this somehow (if this is the case, what about lottefiles https://lottiefiles.com/)?

still neat tho

File: src/Components/Background/background.jsx (hunk 1 above)
Comment (Luceium) [reply src/Components/Background/background.jsx:18]: Great Idea. For now I've extracted them to be global variables instantiated at the top of the file.

File: src/Components/Background/background.jsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        <g>
+          <g transform={`translate(${size / 2}, ${size / 2})`}>
+            <g id="boxes" transform="rotate(45 0 0)">
+              <rect x={-1 * outerRadius} y={-1 * outerRadius} width={2 * outerRadius}
+                height={2 * outerRadius} rx={radius} strokeWidth={size * 0.004}/>
+              <rect x={-1 * innerRadius} y={-1 * innerRadius} width={2 * innerRadius}
+                height={2 * innerRadius} rx={radius} strokeWidth={size * 0.004}/>
+            </g>
+            <g id="circles">
+              <circle cx={lineEnd + circleRadius} cy={-lineMid - spread}
```
[the remark this answers is not in the document]
Comment (Luceium) [reply src/Components/Background/background.jsx:78]: Yes, these calculations are all closely inter-related. The center of each circle is at one of two distances and is either positive or negative. One observation is that the spread is intended to incresase the distance so each coordinate can be written as `sign(distance+spread)`. Would you recommend making a helper functions that maps an array to all the elements. For example `[xSign, xDistance, ySign, yDistance] --> <circle cx={xSign(xDistance + spread)} cy={ySign(yDistance + spread)} />`

[the remark this answers is not in the document]
Comment (Luceium) [reply src/Components/Background/background.jsx:78]: Oh, I see that's what you were talking about it the next comment. I'll get on it soon. 👍🏽

**Comment by Luceium** [comment]
@hsorathia
1 & 2. Yes, but the blue lines are only intended to be added to the circuit lines
3. Yes, the plan is to have an animated background. I looked into lottie Files a bit and I can't find if there's a way to dynamically change it. Ex. I made the component to have many optional props to change the appearance and other factors.
We may also incorporate javascript in the future to make the circuits shoot out randomly.
Is it possible to use scripts and code with lottie files?
If not, maybe I can make a lottie file that has pseudoRandom circuit firing built in.

**Review by akshtshrma24 — COMMENTED** [review]
You have teh same lines changed in 2 PRs this will be bad, merge the OS dependent hot reload, close this PR, open a new PR that has the most recent changes

**Comment by Luceium** [comment]
I merged the OS dependent hot reload and then merged the change into this PR.

**Comment by hsorathia** [comment]
> @hsorathia 1 & 2. Yes, but the blue lines are only intended to be added to the circuit lines 3. Yes, the plan is to have an animated background. I looked into lottie Files a bit and I can't find if there's a way to dynamically change it. Ex. I made the component to have many optional props to change the appearance and other factors. We may also incorporate javascript in the future to make the circuits shoot out randomly. Is it possible to use scripts and code with lottie files? If not, maybe I can make a lottie file that has pseudoRandom circuit firing built in.

I'm unsure if you can randomize them via scripting. I think they're meant to be lightweight static animations. I haven't used them extensively before but this usecase seemed similar enough. If you need the random effects to shoot out I think having javascript done the way you have it might be fine. Or, you can just make an animation that is long enough where it seems random (a few seconds long). 

either way is fine tho bc this is just a log in page

**Comment by Luceium** [comment]
Thanks @hsorathia, I think I'll look into this more in the future. As of right now, converting my react to a usable vector for Lottie files is taking too much time and effort.

**Review by akshtshrma24 — APPROVED** [review]
Now we are talkin

**Pull request merged**